### DISCUSSION TITLE
Does Nebari require node group names to be "general", "user", and "worker"?

### DISCUSSION THREAD

**Discussion by Adam-D-Lewis** [nebari-dev/nebari#2816] [General]
Does Nebari require node group names to be "general", "user", and "worker" or do we support arbitrary node group names?  Stated differently, does a deployment have to have a "general", and "user" node group?  I think I've heard of some users renaming some node groups in the past.  I'm considering enforcing the "general", "user", "worker" node group names a bit in a PR I'm working on and wondering how big of an issue it'll be.

BTW, I don't have an issue with continuing to support additional arbitrary node group names beyond "general", "user", and "worker".  My question is mostly around guaranteeing "general", "user", and "worker" node group names exist.

I did see this guide on pre-existing deployments says those should be the node group names -  https://www.nebari.dev/docs/how-tos/nebari-kubernetes#creating-node-groups

**Comment by marcelovilla** [comment]
I'm not sure how strictly we're enforcing having `general`, `user`, and `worker` being the node group names. For example, in AWS we have `general` hardcoded in several locations:

https://github.com/nebari-dev/nebari/blob/9b1310b33e89c2c11c3b39128ec792ca80342486/src/_nebari/stages/infrastructure/template/aws/modules/kubernetes/main.tf#L135

https://github.com/nebari-dev/nebari/blob/9b1310b33e89c2c11c3b39128ec792ca80342486/src/_nebari/stages/infrastructure/template/aws/modules/kubernetes/main.tf#L156

https://github.com/nebari-dev/nebari/blob/9b1310b33e89c2c11c3b39128ec792ca80342486/src/_nebari/stages/kubernetes_initialize/template/modules/cluster-autoscaler/main.tf#L32

In other places, it seems we're using variables to refer to them:

https://github.com/nebari-dev/nebari/blob/9b1310b33e89c2c11c3b39128ec792ca80342486/src/_nebari/stages/kubernetes_ingress/template/main.tf#L6

I haven't tested changing the node group names of any of the three default names but I wouldn't be surprised if things failed while deploying Nebari. I also haven't seen anyone using different names for those three groups yet. I guess we could verify if that's even possible.

It makes sense to me to enforce those names for the default group nodes, because (1) we'll be consistent across deployments and use the same terminology, an (2) I don't see any benefit in having custom names for them. That being said, it would not be bad to still get rid of the hardcoded names in our Terraform configurations and use variables instead.

**Comment by Adam-D-Lewis** [comment]
Thanks for that info, @marcelovilla. I think it's likely that it is a requirement though not enforced as it ideally should be.  I suspect Nebari won't function properly if alternative names are used for the node groups.  I think it's a reasonable requirement to have, and nothing prevents us from abstracting the name out if we choose to do so at a later date.
